### PR TITLE
feature(ci): add github actions

### PR DESCRIPTION
**Pull request by doanguyen**

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+python3 generate_config.py --certbot && certbot -n certonly; crond && ./generate_config.py --postfix && postfix start-fg
+
+# Idea taken from https://github.com/Mailu/Mailu/blob/master/core/postfix/Dockerfile
+HEALTHCHECK --start-period=350s CMD echo QUIT|nc localhost 25|grep "220 .* ESMTP Postfix"
```
Comment (nguyenkims) [inline entrypoint.sh:6]: should this HEALTHCHECK be moved into the Dockerfile?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          dockerfile: Dockerfile
+      -
+        name: Login to GitHub Container Registry
+        uses: docker/login-action@v1 
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GH_PACKAGE_TOKEN }}
+      -
+        name: Build and push
```
Comment (nguyenkims) [inline .github/workflows/main.yml:26]: can we the action used in the `app` repo https://github.com/simple-login/app/blob/c61dd9dec6460eaa84c55b44e437d1c6991fd3a0/.github/workflows/main.yml#L52 ?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      - uses: actions/checkout@v2
+      - uses: brpaz/hadolint-action@v1.2.1
+        name: Lint
+        with:
+          dockerfile: Dockerfile
+      -
+        name: Login to GitHub Container Registry
+        uses: docker/login-action@v1 
+        with:
+          registry: ghcr.io
```
Comment (nguyenkims) [inline .github/workflows/main.yml:22]: I think we can stick with the default Docker hub for now as this doesn't require an additional step of `docker login`

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: brpaz/hadolint-action@v1.2.1
```
Comment (nguyenkims) [inline .github/workflows/main.yml:14]: 👌

**Review by nguyenkims — COMMENTED** [review]
Looks good! Just left some comments :)

File: entrypoint.sh (hunk 1 above)
Comment (doanguyen) [reply entrypoint.sh:6]: I thought it served the same purpose when starting the container?

Comment (nguyenkims) [reply entrypoint.sh:6]: ah `HEALTHCHECK` is only used inside a Dockerfile and it isn't recognized in a bash script.

**Comment by doanguyen** [comment]
@nguyenkims I think the CI is passed regarding the results because we don't have the `secrets` to login to dockerhub

**Comment by nguyenkims** [comment]
> @nguyenkims I think the CI is passed regarding the results because we don't have the `secrets` to login to dockerhub

Hey I have setup the Docker Hub secrets for this repo, hopefully the CI would pass now :)

**Pull request merged**